`StandaloneEdgeRouterConfigTest.test_48_router_in_error` does not give router enough time

On a busy machine, the test may start reading the router log before the router finished starting up, parsing the config, and getting a chance to log an error.

https://github.com/skupperproject/skupper-router/runs/5575764409?check_suite_focus=true#step:9:4848

```
18:         test_pass = False
18:         with open(self.router_5.outfile + '.out', 'r') as out_file:
18:             for line in out_file:
18:                 if "Exception: Cannot load configuration file test-router-5.conf: role='standalone' not allowed to connect to or accept connections from other routers." in line:
18:                     test_pass = True
18:                     break
18: >       self.assertTrue(test_pass)
18: E       AssertionError: False is not true
18: 
18: /home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_one_router.py:299: AssertionError
18: - generated xml file: /home/runner/work/skupper-router/skupper-router/skupper-router/build/tests/junitxmls/system_tests_one_router.xml -
18: =========================== short test summary info ============================
18: FAILED ::StandaloneEdgeRouterConfigTest::test_48_router_in_error - AssertionE...
18: =================== 1 failed, 50 passed in 187.94s (0:03:07) ===================
29/31 Test #18: system_tests_one_router ...............***Failed  193.94 sec
```

The following code in the test is insufficient protection against this situation

```python
        # Give some time for the test to write to the .out file. Without
        # this sleep, the tests execute too
        # fast and find that nothing has yet been written to the .out files.
        sleep(3)
```

--- a/tests/system_test.py
+++ b/tests/system_test.py
@@ -883,7 +883,7 @@ class Tester:
         """Start a Process that will be cleaned up on teardown"""
         return self.cleanup(Process(*args, **kwargs))
 
-    def qdrouterd(self, *args, **kwargs):
+    def qdrouterd(self, *args, **kwargs) -> Qdrouterd:
         """Return a Qdrouterd that will be cleaned up on teardown"""
         return self.cleanup(Qdrouterd(*args, **kwargs))
 

--- a/tests/system_tests_one_router.py
+++ b/tests/system_tests_one_router.py
@@ -21,6 +21,7 @@ import json
 import os
 from subprocess import PIPE, STDOUT
 from time import sleep
+from typing import Sequence
 
 from proton import Condition, Delivery, Message, Timeout, symbol
 from proton.handlers import MessagingHandler
@@ -197,7 +198,7 @@ class StandaloneEdgeRouterConfigTest(TestCase):
             ('router', {'mode': 'standalone', 'id': 'QDR'}),
             ('listener', {'port': cls.tester.get_port(), 'role': 'edge', 'host': '0.0.0.0'})
         ])
-        cls.router = cls.tester.qdrouterd(name, config, wait=False, perform_teardown=False)
+        cls.router = cls.tester.qdrouterd(name, config, wait=False, expect=Process.EXIT_FAIL)
 
         # A standalone router cannot have inter-router connectors.
         name = "test-router-1"
@@ -205,7 +206,7 @@ class StandaloneEdgeRouterConfigTest(TestCase):
             ('router', {'mode': 'standalone', 'id': 'QDR'}),
             ('connector', {'port': cls.tester.get_port(), 'role': 'inter-router', 'host': '0.0.0.0'})
         ])
-        cls.router_1 = cls.tester.qdrouterd(name, config_1, wait=False, perform_teardown=False)
+        cls.router_1 = cls.tester.qdrouterd(name, config_1, wait=False, expect=Process.EXIT_FAIL)
 
         # An edge router cannot have edge listeners.
         # Edge routers can have connectors that connect to interior routers
@@ -215,7 +216,7 @@ class StandaloneEdgeRouterConfigTest(TestCase):
             ('router', {'mode': 'edge', 'id': 'QDR'}),
             ('listener', {'port': cls.tester.get_port(), 'role': 'edge', 'host': '0.0.0.0'})
         ])
-        cls.router_2 = cls.tester.qdrouterd(name, config_2, wait=False, perform_teardown=False)
+        cls.router_2 = cls.tester.qdrouterd(name, config_2, wait=False, expect=Process.EXIT_FAIL)
 
         # Edge routers cannot have inter-router listeners. Only interior
         # routers can have inter-router listeners.
@@ -224,7 +225,7 @@ class StandaloneEdgeRouterConfigTest(TestCase):
             ('router', {'mode': 'edge', 'id': 'QDR'}),
             ('listener', {'port': cls.tester.get_port(), 'role': 'inter-router', 'host': '0.0.0.0'})
         ])
-        cls.router_3 = cls.tester.qdrouterd(name, config_3, wait=False, perform_teardown=False)
+        cls.router_3 = cls.tester.qdrouterd(name, config_3, wait=False, expect=Process.EXIT_FAIL)
 
         # Edge routers cannot have inter-router connectors
         # Inter-router connectors are allowed only on interior routers.
@@ -233,7 +234,7 @@ class StandaloneEdgeRouterConfigTest(TestCase):
             ('router', {'mode': 'edge', 'id': 'QDR'}),
             ('connector', {'port': cls.tester.get_port(), 'role': 'inter-router', 'host': '0.0.0.0'})
         ])
-        cls.router_4 = cls.tester.qdrouterd(name, config_4, wait=False, perform_teardown=False)
+        cls.router_4 = cls.tester.qdrouterd(name, config_4, wait=False, expect=Process.EXIT_FAIL)
 
         # A standalone router cannot have an inter-router listener because
         # it cannot accept inter-router connections.
@@ -242,12 +243,13 @@ class StandaloneEdgeRouterConfigTest(TestCase):
             ('router', {'mode': 'standalone', 'id': 'QDR'}),
             ('listener', {'port': cls.tester.get_port(), 'role': 'inter-router', 'host': '0.0.0.0'})
         ])
-        cls.router_5 = cls.tester.qdrouterd(name, config_5, wait=False, perform_teardown=False)
+        cls.router_5 = cls.tester.qdrouterd(name, config_5, wait=False, expect=Process.EXIT_FAIL)
 
-        # Give some time for the test to write to the .out file. Without
-        # this sleep, the tests execute too
+        # Give some time for the test to write to the .out file. Without this, the tests execute too
         # fast and find that nothing has yet been written to the .out files.
-        sleep(3)
+        routers: Sequence[Qdrouterd] = (cls.router, cls.router_1, cls.router_2, cls.router_3, cls.router_4, cls.router_5)
+        for router in routers:
+            router.wait(timeout=TIMEOUT)
 
     def test_48_router_in_error(self):
         test_pass = False

